### DISCUSSION TITLE
Duplicate / Identical Auto ID from Row Creation in Prod

### DISCUSSION THREAD

**Discussion by EmonSM** [Budibase/budibase#9269] [Help]
Hi,

I noticed a serious issue with the Auto ID column in my form: duplicate numbers are frequently being assigned to the Auto ID field, even though the autogenerated field promises uniqueness. 

See screenshots:
![Screen Shot 2023-01-08 at 10 00 19 AM](https://user-images.githubusercontent.com/55682406/211205720-02eec2c2-836b-4bb0-8c13-6855f7b7210e.png)
![Screen Shot 2023-01-08 at 9 59 49 AM](https://user-images.githubusercontent.com/55682406/211205721-8abd997b-e95a-4268-aba6-e6e378e2bf7f.png)

I'm aware that the same issue was discussed in this thread https://github.com/Budibase/budibase/discussions/7105, but the root cause of my issue must be completely different, as I verified that the duplicate Auto ID rows were all created in prod (as the users responsible did not have access to dev). Therefore, these duplicate Auto ID values cannot be due to the same prod & dev sync issue identified in that thread. 

Please look into why this is occurring, as my Budibase integration relies on this Auto ID field being a unique identifier for each row. 

If the cause can't be identified, is there a way to always force an increment somehow? It's fine for me if there's potentially some values skipped in such an approach, I only need uniqueness guaranteed.

Thank you in advance!

**Comment by melohagan** [comment]
Hey @EmonSM 

Would you be able to share your app export to: support@budibase.com ?

Do you have automations that create rows into this table? Just a thought.

**Comment by EmonSM** [comment]
Hey @melohagan,

Thanks for the quick response. No, we don't have any automations that create rows into this table. All of the duplicate Auto ID rows were manually created by users in the prod app. The typical scenario was that the same individual would create two different form submissions minutes apart and the Auto ID was duplicated for some reason.

We will follow your suggestion and share the app export to [support@budibase.com](mailto:support@budibase.com).

Thanks,
Emon

**Comment by applicationuser** [comment]
Hello @melohagan 

we also noticed similar issue , initial table data was loaded through CSV import.

Later created an automation to update and add rows.   On Automation execute, Create row automation step added row that had duplicate Auto ID value 

Budibase Cloud 
app version : 2.2.22
App not published yet,  error noticed in development mode

![image](https://user-images.githubusercontent.com/42922833/215115387-b6699851-6315-4275-987f-e90d65bf2364.png)

**Comment by DustinDobernig** [comment]
Had the same issue happen today. Not certain what caused it. My app is all based on this functionality as well.

2.7.26.
Self Hosted

**Comment by Shadowfita** [comment]
I've experienced this as well.

**Reply by Shadowfita** [reply]
@melohagan  is there a way to update the AutoIDs? This is a pretty breaking issue for some data structures in my app.

**Reply by melohagan** [reply]
One way could be to create an automation that queries rows (sorting on the auto ID ascending) and then doing an update / delete row action on loop to merge rows.

Alternatively if you want to keep both rows that have a duplicate Auto ID, then your automation could create a new row for each duplicate, and then delete that duplicate.

**Comment by zan-xhipe** [comment]
<img width="812" height="384" alt="Screenshot 2025-08-13 at 15-23-35 Budibase" src="https://github.com/user-attachments/assets/e0d5f55e-6e12-4259-b100-799bd4f31eef" />


I am also getting duplicate AutoIDs all the entries are created manually. Screenshot shows the auto updated at and auto id columns
